TypeCobol.Test/Parser/Programs/Cobol85/WhenTest.rdzMix.txt

```diff
@@ -0,0 +1,48 @@
+       PROGRAM-ID. WhenTest.
+       data division.
+       working-storage section.
+       01 var1 pic X(5) value 'hi'.
+       01 var2 pic X(5) value 'ho'.
+       01 var3 pic X(5) value 'ho'.
+       01 bool1 pic 9.
+       01 bool2 pic 9.
+       PROCEDURE DIVISION.
+           evaluate true
+Line 12[17,20] <27, Error, Syntax> - Syntax error : Missing condition in "when" clause
+                when
+                when bool1 display "evaluate"
+                when bool1
+                     when bool2 display "evaluate"
+Line 16[17,20] <27, Error, Syntax> - Syntax error : Missing condition in "when" clause
+Line 16[17,20] <27, Error, Syntax> - Syntax error : Missing statement in "when" clause
+                when
+                when other continue
+           end-evaluate
+           evaluate true
+                when bool1 display "evaluate"
+Line 21[17,26] <27, Error, Syntax> - Syntax error : Missing statement in "when" clause
+                when bool1
+                when other continue
+           end-evaluate      
+Line 24[12,22] <27, Error, Syntax> - Syntax error : Cannot SEARCH in 'var1', data item is not a table.
+           search var1
+Line 25[17,20] <27, Error, Syntax> - Syntax error : Missing condition in "when" clause
+Line 25[17,20] <27, Error, Syntax> - Syntax error : extraneous input 'when' expecting {AtEndCondition, WhenSearchCondition}
+                when
+                when var2
+                    when var3 display "search"
+Line 28[17,20] <27, Error, Syntax> - Syntax error : Missing condition in "when" clause
+                when
+           end-search
+Line 30[12,24] <37, Warning, General> - Warning: "when other" is missing
+           evaluate true
+                when bool1 display "evaluate"
+      *KO missing statement in "when other" clause
+                when other
+Line 34[12,23] <27, Error, Syntax> - Syntax error : extraneous input 'end-evaluate' expecting {AcceptStatement, AddStatement, AllocateStatement, AlterStatement, CallStatement, CancelStatement, CloseStatement, ComputeStatement, ContinueStatement, DeleteStatement, DisplayStatement, DivideStatement, EntryStatement, EvaluateStatement, ExecStatement, ExitMethodStatement, ExitProgramStatement, ExitStatement, FreeStatement, GobackStatement, GotoStatement, IfStatement, InitializeStatement, InspectStatement, InvokeStatement, JsonGenerateStatement, JsonParseStatement, MergeStatement, MoveStatement, MultiplyStatement, OpenStatement, PerformProcedureStatement, PerformStatement, ReadStatement, ReleaseStatement, ReturnStatement, RewriteStatement, SearchStatement, SetStatement, SortStatement, StartStatement, StopStatement, StringStatement, SubtractStatement, UnstringStatement, WriteStatement, XmlGenerateStatement, XmlParseStatement, ProcedureStyleCall}
+           end-evaluate      
+           move "hi" to var1
+           goback.
+Line 37[8,28] <27, Error, Syntax> - Syntax error : Unexpected orphan "PROGRAM END".
```

Comment:
Is this error due to the previous empty `when other`?
It's just to be sure that an empty `when` won't produce unexpected error.

Reply:
Yes the end's parent is in this situation is a sentence